clippy_lints/src/needless_pass_by_value.rs

```diff
@@ -340,3 +335,22 @@ impl<'tcx> euv::Delegate<'tcx> for MovedVariablesCtxt {
 
     fn fake_read(&mut self, _: &rustc_hir_typeck::expr_use_visitor::PlaceWithHirId<'tcx>, _: FakeReadCause, _: HirId) {}
 }
+
+/// Peel `Option<…>` from `hir_ty` as long as the HIR name is `Option` and it corresponds to the
+/// `core::Option<_>` type.
+fn peel_options<'tcx>(
```

Comment:
Seems like this will be useful in clippy utils. Can we have it there?

Reply:
Done. I have also added two extra commits while doing the move:
- one to mark the applicability as unspecified, as we are suggesting modifying a function parameter without fixing the call sites
- one to make the suggestion verbose, which reduces the diagnostic readability

Those are different commits because they fix or enhance unrelated concerns, let me know if you prefer them merged, or omitted for the verbose suggestion one.